src/entities/EditorDocument/EditorDocument.spec.ts

```diff
@@ -0,0 +1,203 @@
+    });
+
+    it('should add the block to the specified index', () => {
+      // Arrange
+      const block = createBlock({
+        name: createBlockNodeName('block'),
+        parent: document,
+      });
+
+      // Act
+      document.addBlock(block, 1);
+
+      // Assert
+      expect(document.getBlock(1)).toBe(block);
+    });
+
+    it('should add the block to the end of the document', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      document.addBlock(block, blocksInDocument);
+
+      // Assert
+      expect(document.getBlock(blocksInDocument)).toBe(block);
+    });
+
+    it('should throw an error if index is greater then block nodes length', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      const action = (): void | never => document.addBlock(block, blocksInDocument + 1);
+
+      // Assert
+      expect(action).toThrowError('Index out of bounds');
+    });
+
+    it('should throw an error if index is less then 0', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      const action = (): void | never => document.addBlock(block, -1);
+
+      // Assert
+      expect(action).toThrowError('Index out of bounds');
+    });
+  });
+
+  describe('removeBlock', () => {
+    it('should remove the block from the beginning of the document', () => {
+      // Arrange
+      const block = document.getBlock(0);
+
+      // Act
+      document.removeBlock(0);
+
+      // Assert
+      expect(document.getBlock(0)).not.toBe(block);
+    });
+
+    it('should remove the block from the specified index', () => {
+      // Arrange
+      const block = document.getBlock(1);
+
+      // Act
+      document.removeBlock(1);
+
+      // Assert
+      expect(document.getBlock(1)).not.toBe(block);
+    });
+
+    it('should remove the block from the end of the document', () => {
+      // Act
+      document.removeBlock(blocksInDocument - 1);
+
+      // Assert
+      expect(() => document.getBlock(blocksInDocument - 1)).toThrowError('Index out of bounds');
+    });
+
+    it('should throw an error if index is greater then block nodes length', () => {
+      // Act
+      const action = (): void | never => document.removeBlock(blocksInDocument);
+
+      // Assert
+      expect(action).toThrowError('Index out of bounds');
+    });
+
+    it('should throw an error if index is less then 0', () => {
+      // Act
+      const action = (): void | never => document.removeBlock(-1);
+
+      // Assert
+      expect(action).toThrowError('Index out of bounds');
+    });
+  });
+
+  describe('getBlock', () => {
+    it('should return the block from the specific index', () => {
+      // Arrange
+      const index = 1;
+
+      // Act
+      const block = document.getBlock(index);
+
+      // Assert
+      expect(block).toBe(document.getBlock(index));
```

Comment:
This test doesn't make any sense. You need to get expected value from somewhere else (from beforeEach hook for example)